servers/publikator/script/lib/transformPublications.js

```diff
@@ -0,0 +1,196 @@
+      sort: {
+        key: 'publishedAt',
+        direction: 'DESC'
+      }
+    }, context)
+
+    pageInfo = docsConnection.pageInfo
+
+    await Promise.each(
+      docsConnection.nodes
+        .filter(node => node.type === 'Document')
+        .map(node => node.entity),
+      (entity) =>
+        callback({ doc: entity }), // eslint-disable-line standard/no-callback-literal
+      { concurrency: 1 }
+    )
+  } while (pageInfo && pageInfo.hasNextPage)
+}
+
+module.exports = ({ transform }) =>
+  PgDb.connect().then(async pgdb => {
+    console.log('running...')
+    const { dry } = argv
+    if (dry) {
+      console.log("dry run: this won't change anything")
+    }
+    const stopAfterNumFixes = argv.n > 0 && argv.n
+    if (stopAfterNumFixes) {
+      console.log(`stop after num fixes: ${stopAfterNumFixes}`)
+    }
+
+    const context = getContext({
+      pgdb,
+      elastic: Elastic.client(),
+      redis,
+      pubsub,
+      t
+    })
+
+    let numFixed = 0
+    let finishNotice = false
+    await iterateESDocs(context, async ({doc}) => {
+      if (stopAfterNumFixes && numFixed >= stopAfterNumFixes) {
+        if (!finishNotice) {
+          console.log(`finish here because stopAfterNumFixes: ${stopAfterNumFixes}`)
+          finishNotice = true
+        }
+        return
+      }
+      let docTransformed = await transform(doc, context)
+      if (docTransformed) {
+        console.log(`\nfixing: ${docTransformed}...`)
+        console.log(`${FRONTEND_BASE_URL}${doc.meta.path}`)
+
+        const parsedDocId = Buffer.from(doc.id, 'base64').toString('utf-8')
+        const [
+          repoOrg, // eslint-disable-line no-unused-vars
+          repoName,
+          commitId,
+          versionName
+        ] = parsedDocId.split('/')
+
+        // for copied repos
+        const repoId = `${GITHUB_LOGIN}/${repoName}`
+        console.log(`https://github.com/${repoId}`)
+
+        if (versionName.indexOf('prepublication') > -1) {
+          console.log('encountered prepublication: skipping', parsedDocId)
+          return
+        }
+
+        // get commit from github instead of ES and do the work again
+        const rawDoc = await getRawDoc(
+          {
+            id: commitId,
+            repo: { id: repoId }
+          },
+          { publicAssets: false },
+          context
+        )
+
+        docTransformed = await transform(rawDoc, context)
+        if (!docTransformed) {
+          console.log('es <> git out of sync!', repoId)
+          return
+        }
+
+        if (dry) {
+          console.log('dry: no commit/publish')
+        } else {
+          console.log('committing...')
+          const newCommit = await commit(null, {
+            repoId,
+            parentId: commitId,
+            message: docTransformed,
+            document: {
+              content: rawDoc.content
+            }
+          }, context)
+
+          console.log('publishing...')
+          const publication = await publish(null, {
```

Comment:
Does this script fall apart if a document was only prepublished and would publish it for good?

Reply:
Yes indeed, good catch. I added `ignorePrepublished` accordingly.